WcaOnRails/app/webpacker/components/RolesTab/index.jsx

```diff
@@ -27,105 +21,67 @@ import { groupTypes, delegateRegionsStatus } from '../../lib/wca-data.js.erb';
-    if (role.group.group_type === groupTypes.delegate_regions) {
-      if ([
-        delegateRegionsStatus.senior_delegate,
-        delegateRegionsStatus.regional_delegate,
-      ].includes(role.metadata.status)) {
-        return panelUrls.board.regionsManager;
-      }
-      return null;
-    }
-    if (role.group.group_type === groupTypes.teams_committees) {
-      return `${teamUrl(role.group.id.split('_').pop())}/edit`;
-    }
-    if (role.group.group_type === groupTypes.translators) {
-      return panelUrls.wst.translators;
-    }
-    return null;
-  }
-
-  function isHyperlinkableRole(role) {
-    if (role.group.group_type === groupTypes.delegate_regions) {
-      return [
-        delegateRegionsStatus.senior_delegate,
-        delegateRegionsStatus.regional_delegate,
-      ].includes(role.metadata.status);
-    }
-    return [groupTypes.teams_committees, groupTypes.translators].includes(role.group.group_type);
-  }
-
-  function getRoleDescription(role) {
-    let roleDescription = '';
-    if (role.metadata.status) {
-      roleDescription += `${I18n.t(`enums.user.role_status.${role.group.group_type}.${role.metadata.status}`)}, `;
-    }
-    roleDescription += role.group.name;
-    return roleDescription;
+  function hasAtLeastOneRole() {
+    return activeRoles?.length > 0 || pastRoles?.length > 0;
   }
 
-  if (roleListFetch.loading || loading) return <Loading />;
-  if (roleListFetch.error) return <Errored />;
+  if (activeRolesLoading || pastRolesLoading) return <Loading />;
+  if (activeRolesError || pastRolesError) return <Errored />;
 
   return (
     <>
-      {roleListFetch.data?.length > 0
+      {hasAtLeastOneRole()
         ? (
           <>
-            <Header>Active Roles</Header>
-            <List divided relaxed>
-              {roleListFetch.data?.map((role) => (
-                <List.Item key={role.id}>
-                  <List.Content
-                    floated="left"
-                    href={hyperlink(role)}
-                  >
-                    <Icon
-                      name="edit"
-                      size="large"
-                      link
-                      disabled={!loggedInUserPermissions.canEditRole(role)}
-                      onClick={isHyperlinkableRole(role) ? null : () => setOpen(true)}
-                    />
-                  </List.Content>
-                  <List.Content>
-                    <List.Header>
-                      {getRoleDescription(role)}
-                    </List.Header>
-                    {!!role.start_date && (
-                      <List.Description>
-                        {`Since ${role.start_date}`}
-                      </List.Description>
-                    )}
-                  </List.Content>
-                </List.Item>
-              ))}
-            </List>
+            {activeRoles?.length > 0 && (
+            <ActiveRoles
+              activeRoles={activeRoles}
+              setOpen={setOpen}
+            />
```

Comment:
Should be indented.

Reply:
Made it in single line as it is now small enough to fit within recommended number of characters of airbnb-eslint (I think 80 or 100)